rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,424 @@
+    uid=globals()['rdkit_wg_dict'][model_id]
+    
+    globals()['view_'+uid].removeAllModels()
+    globals()['view_'+uid].removeAllSurfaces()
+    globals()['view_'+uid].removeAllLabels()
+    
+    
+    molDictKey=globals()['molTupleId_'+uid].value
+    
+    if globals()['selectAllMols_'+uid].value is True:
+        keys=list(globals()['moldict_'+uid].keys())
+        for i in keys:
+            globals()['rdkit_mol_select_'+uid].append(i)
+    else:
+        if globals()['selectMultiMols_'+uid].value is True:
+            globals()['rdkit_mol_select_'+uid].append(molDictKey)
+        else:
+            globals()['rdkit_mol_select_'+uid] = list()
+            globals()['rdkit_mol_select_'+uid].append(molDictKey)
+    
+    
+    molNameList = list(set(globals()['rdkit_mol_select_'+uid]))
+    globals()['selected_mols_view_'+uid].value = ', '.join(molNameList)
+    
+    
+    for i in molNameList:
+        mol = globals()['moldict_'+uid][i]
+        
+        if mol.GetNumConformers()>1:
+            
+            allConfIds = range(mol.GetNumConformers()-1)
+            globals()['confId_'+uid].options = allConfIds
+            
+            
+            if globals()['selectAllConfs_'+uid].value is True:
+                for i in allConfIds:
+                    globals()['rdkit_conf_select_'+uid].append(i)
+            else:
+                confId=globals()['confId_'+uid].value
+                if globals()['selectMultiConfs_'+uid].value is True:
+                    globals()['rdkit_conf_select_'+uid].append(confId)
+                else:
+                    globals()['rdkit_conf_select_'+uid] = list()
+                    globals()['rdkit_conf_select_'+uid].append(confId)
+                    
+            
+        elif mol.GetNumConformers()==1:
+            globals()['confId_'+uid].options=[0]
+            confId=globals()['confId_'+uid].value
+            globals()['rdkit_conf_select_'+uid]=list()
+            globals()['rdkit_conf_select_'+uid].append(confId)
+            globals()['selected_confs_view_'+uid].value = str(confId)
+        
+        
+    confIdsList = list(set(globals()['rdkit_conf_select_'+uid]))
+    globals()['selected_confs_view_'+uid].value = ', '.join([str(x) for x in confIdsList])
+    
+    
+    # Add molecule to viewer
+    for molName in molNameList:
+        for confId in confIdsList:
+            mol = globals()['moldict_'+uid][molName]
+            mb = Chem.MolToMolBlock(mol,confId=confId)
+            globals()['view_'+uid].addModel(mb,'sdf')
+    
+    
+    if len(molNameList)==1:
+        mol = globals()['moldict_'+uid][molDictKey]
+        # For precalculated property
+        try:
+            all_prop_from_mol=list(mol.GetPropNames())
+            if len(all_prop_from_mol)>0:
+                # update widget
+                globals()['prop_precalc_wg_'+uid].options=all_prop_from_mol
+                # Get selected property
+                prop_name=eval('prop_precalc_wg_'+uid+'.value')
```

Comment:
I was totally unaware about dangerousness of eval. Extremely sorry for the bad code. This has been corrected. And the most recent commit should make sense.